src/Kestrel.Core/KestrelServerLimits.cs

```diff
@@ -251,6 +252,25 @@ public TimeSpan RequestHeadersTimeout
             }
         }
 
+        /// <summary>
+        /// Limits the number of concurrent request streams per HTTP/2 connection. Excess streams will be refused.
+        /// <para>
+        /// Defaults to 100
+        /// </para>
+        /// </summary>
+        public int MaxStreamsPerConnection
+        {
+            get => _maxStreamsPerConnection;
+            set
+            {
+                if (value < 0)
```

Comment:
`<=`

Reply:
0 is technically allowed. You can have the endpoint up for pings but not accept requests.

Reply:
Why though? Even if if allowed by protocol, we don't need to allow a pit of failure.